timings.c

```diff
@@ -76,7 +97,33 @@ void spy(char **addrs, size_t num_addrs, int num_iterations) {
         }
     }
 
-    printf("Statistics after %d runs:\n", num_iterations);
+    printf("Statistics for memory after %d runs:\n", num_iterations);
+    for (size_t addr = 0; addr < num_addrs; addr++) {
+        printf("Address #%d statistics:\n", (int) addr);
+        printf("min = %lu\n", addr_stats[addr].min);
+        printf("max = %lu\n", addr_stats[addr].max);
+        printf("mean = %lu\n", addr_stats[addr].sum / num_iterations);
+    }
+
+    // init for cache read
+    for (size_t addr = 0; addr < num_addrs; addr++) {
+        addr_stats[addr].min = (unsigned long) -1;
```

Comment:
Let's change this to `addr_stats[addr].min = ULONG_MAX;` to avoid invoking technically undefined behavior. This probably requires adding an `#include <stdint.h>` or `#include <limits.h>`.

(Unfortunately we weren't that great at writing C 1.5 years ago either.)